freqtrade/persistence/models.py

```diff
@@ -278,6 +306,22 @@ def open_date_utc(self):
     def close_date_utc(self):
         return self.close_date.replace(tzinfo=timezone.utc)
 
+    def __init__(self, **kwargs):
+        if kwargs.get('leverage') and kwargs.get('borrowed'):
+            # TODO-mg: should I raise an error?
+            raise OperationalException('Cannot pass both borrowed and leverage to Trade')
```

Comment:
I think this is a theoretical problem (possibly relevant for tests - but not for the rest).
Trade objects are not created by the user, but by Freqtrade code only - so if the trade creation (only happens in `freqtradebot.py` and `backtesting.py` ) is done properly, this should never be the case.

Reply:
What should never be the case? Do you that this branch should never be visited?

Reply:
Well this is the `__init__` method (constructor). As such - freqtrade (speak - we) controll what is passed in here - so hitting this branch will be a programming error we didn't encounter (not a problem in the strategy / user error) - and as such, i (hope) it's a theoretical problem :laughing:

Reply:
yeah the purpose of that conditional is to catch programming errors

Reply:
maybe add a comment that makes that clear ... otherwise i might have the same doubt in a few days (or months ...)

Reply:
Is `OperationalException` meant for strategy / user error and is there another error or exception type which would make it clear that it's to prevent a programmer error?

Reply:
OperationalException is fine - it's the most severe error we have - but it's the only one intended to "stop bot execution" - which i think is fine for a programmer error.

We've the exception hierarchy [documented](https://www.freqtrade.io/en/stable/developer/#errorhandling) as well should you need one of the others for whatever reason.